torch/csrc/jit/passes/liveness.cpp

```diff
@@ -82,38 +89,41 @@ struct LivenessAnalyzer {
       // kill outputs
       liveness -= toSparseBitVector(it->outputs());
       if (it->kind() == prim::Loop) {
-        auto loop_block = liveness;
+        LoopView lv(it);
+        // N.B. merge in changes from the loop header
+        auto loop_header = *lv.bodyBlock()->nodes().begin();
+        auto loop_block = liveness | liveness_sets_[loop_header];
         // loop's outputs aren't live inside the loop
         // loop's block outputs, OTOH, will be considered
         // as uses
-
-        LoopView lv(it);
         WithInsertPoint guard(*lv.bodyBlock()->nodes().end());
         // temporary make loop counts live for the duration of the loop
         // as they are needed by BailOuts in the loop
         auto ctc = graph_->create(prim::Store, {lv.currentTripCount()}, 0);
         graph_->insertNode(ctc);
         auto mtc = graph_->create(prim::Store, {lv.maxTripCount()}, 0);
         graph_->insertNode(mtc);
-        loop_block = processBlock(it->blocks()[0], loop_block);
+        loop_block = processBlock(lv.bodyBlock(), loop_block);
         ctc->destroy();
         mtc->destroy();
         // loop block's inputs die outside loop's block
-        loop_block -= toSparseBitVector(it->blocks()[0]->inputs());
+        loop_block -= toSparseBitVector(lv.bodyBlock()->inputs());
         liveness |= loop_block;
       } else if (it->kind() == prim::If) {
-        auto true_liveness = processBlock(it->blocks()[0], liveness);
-        auto false_liveness = processBlock(it->blocks()[1], liveness);
+        IfView iv(it);
+        auto true_liveness = processBlock(iv.thenBlock(), liveness);
+        auto false_liveness = processBlock(iv.elseBlock(), liveness);
         liveness |= true_liveness;
         liveness |= false_liveness;
       }
       liveness |= toSparseBitVector(it->inputs());
-      liveness_sets_.insert({it, liveness});
+      changed_ = changed_ | (liveness_sets_[it] |= liveness);
```

Comment:
Can this be rewritten in a way that makes it more obvious it is calculating that liveness_sets is getting changed? I don't know what the return of a `|=` actually is.